### ISSUE TITLE
PyTorch non-static autograd.Function loss cause issue

### ISSUE THREAD

**Issue by chanil1218** [HawkAaron/warp-transducer#22]
From 339471ac55b204ce8d949c86bd60c2981581505c this commit, 
Loss function changed from static autograd.Function to non-static Function.

With this change, model with non-static Function loss cannot be trained(or properly back-propagated).

Check [forum comment](https://discuss.pytorch.org/t/custom-autograd-function-must-it-be-static/14980/2) that is related to this issue. 

My suggestion is that keep wrapper module with static inner function.

**Comment by HawkAaron** [comment]
Could you provide an example ? I don't understand "_model with non-static Function loss cannot be trained_".

**Comment by chanil1218** [comment]
I found that model did not decrease loss after 339471a commit.
So I reverted it with static autograd.Function(decoradted with `@staticmethod`, (2) in the https://discuss.pytorch.org/t/custom-autograd-function-must-it-be-static/14980), and then work as expected.

![after_339471a](https://user-images.githubusercontent.com/1420133/58161835-fa34be80-7cbb-11e9-8719-6e39b84b5dcb.png)
![before_339471a](https://user-images.githubusercontent.com/1420133/58161848-ff920900-7cbb-11e9-837f-f27ea88e7453.png)

I think

**Comment by HawkAaron** [comment]
Thanks very much! The bug is fixed, please have a try.

**Comment by chanil1218** [comment]
Great! Works as expected. 👍
